dev/cuda/crossentropy_forward.cu

```diff
@@ -42,12 +42,28 @@ __global__ void crossentropy_forward_kernel1(float* losses,
     if (i < B * T) {
         int b = i / T;
         int t = i % T;
-        const float* probs_bt = probs + b * T * V + t * V;
-        int ix = targets[b * T + t];
-        losses[b * T + t] = -logf(probs_bt[ix]);
+        const int bt = b * T + t;
+        const float* probs_bt = probs + bt * V;
+        int ix = targets[bt];
+        losses[bt] = -logf(probs_bt[ix]);
     }
 }
 
+__global__ void crossentropy_forward_kernel2(float* losses,
+    const float* probs_pad, const int* targets,
+    int B, int T, int V, int V_pad) {
+    int i = blockIdx.x * blockDim.x + threadIdx.x;
+    if (i < B * T) {
+        int b = i / T;
+        int t = i % T;
+        const int bt = b * T + t;
+        const float* probs_bt = probs_pad + bt * V_pad;
+        int ix = targets[bt];
+        if (ix < V) { // Make sure we don't read past the end of the original V-sized block
```

Comment:
this should really be an assert, not an if.
If ix is out of range, something went seriously wrong somewhere before.

Reply:
Agreed, removed it and tests running fine.